docs/learn/dapp-developer/read-asset-data.md

```diff
@@ -265,9 +267,674 @@ console.log(assetMetadata);
+
+If neither _Token ID Metadata_ nor _Base URI_ are set, you should fall back and fetch the [Global Token Information](#global-token-information), potentially including a Base URI to [retrieve the JSON content](#get-the-json-content).
+
+:::
+
+To get token ID metadata, you will have to make direct contract calls.Therefore, install a provider library, set up the contract, and import the related contract ABIs. Its also recommended to use the [`erc725.js`](../../tools/erc725js/getting-started.md) and [`lsp-smart-contracts`](../../tools/lsp-smart-contracts/getting-started.md) libraries to easily get [ERC725Y](../../standards/lsp-background/erc725.md) data keys and decode their values.
+
+<Tabs groupId="provider-lib">
+  <TabItem value="ethers" label="ethers">
+
+```bash
+npm install ethers @erc725/erc725.js @lukso/lsp-smart-contracts
+```
+
+```js
+// Add the necessary imports to your JS file
+import { ethers } from 'ethers';
+import lsp8Artifact from '@lukso/lsp-smart-contracts/artifacts/LSP8IdentifiableDigitalAsset.json';
+import { INTERFACE_IDS, ERC725YDataKeys } from '@lukso/lsp-smart-contracts';
+
+const provider = new ethers.JsonRpcProvider(
+  'https://rpc.testnet.lukso.gateway.fm',
+);
+
+// Create contract instance
+const myAssetContract = new ethers.Contract(
+  '<myAssetAddress>',
+  lsp8Artifact.abi,
+  provider,
+);
+```
+
+  </TabItem>
+
+  <TabItem value="web3" label="web3">
+
+```bash
+npm install web3 @erc725/erc725.js @lukso/lsp-smart-contracts
+```
+
+```js
+// Add the necessary imports to your JS file
+import Web3 from 'web3';
+import lsp8Artifact from '@lukso/lsp-smart-contracts/artifacts/LSP8IdentifiableDigitalAsset.json';
+import { INTERFACE_IDS, ERC725YDataKeys } from '@lukso/lsp-smart-contracts';
+
+const web3 = new Web3(
+  new Web3.providers.HttpProvider('https://rpc.testnet.lukso.gateway.fm'),
+);
+// Create contract instance
+const myAssetContract = new web3.eth.Contract(
+  lsp8Artifact.abi,
+  '<myAssetAddress>',
+);
+```
+
+  </TabItem>
+
+</Tabs>
+
+:::caution
+
+Make sure to adjust `<myAssetAddress>` with the actual address of the asset. You can give it a try using a sample LSP8 asset on the LUKSO Testnet: [`0x8734600968c7e7193BB9B1b005677B4edBaDcD18`](https://wallet.universalprofile.cloud/asset/0x8734600968c7e7193BB9B1b005677B4edBaDcD18?network=testnet).
+
+:::
+
+### Preparing the Token IDs
+
+The [LSP8](../../standards/tokens/LSP8-Identifiable-Digital-Asset.md) allows for different [Token ID Formats](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-8-IdentifiableDigitalAsset.md#lsp8tokenidformat), meaning developers can specify their token IDs as:
+
+- `Number` of `uint256` (`TokenIdFormat`: `0` or `100`)
```

Comment:
Please let's not re-write the same thing we have in some page in multiple places. Duplicate content is what will create documentation that is very hard to maintain.

Please just remove these bullet points. Developer can go to the right page to learn more.